### PR TITLE
feat(test): integrate ethereum tests

### PR DESCRIPTION
**Pull request by magicalne**
Integrate [VMTest in ethereum tests](https://github.com/ethereum/tests/blob/develop/GeneralStateTests/VMTests/).

### CODE REVIEW

File: polyjuice-tests/src/ctx.rs [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+    bytes::Bytes,
+    core::{AllowedContractType, AllowedEoaType, ScriptHashType},
+    offchain::{RollupContext, RunResult},
+    packed::{AllowedTypeHash, BlockInfo, RawL2Transaction, RollupConfig, Script, Uint64},
+    prelude::*,
+    U256,
+};
+
+use crate::helper::{PolyjuiceArgsBuilder, POLYJUICE_PROGRAM_CODE_HASH};
+pub const CREATOR_ACCOUNT_ID: u32 = 3;
```
Comment (Flouse) [inline polyjuice-tests/src/ctx.rs:43]: Why not use the variables defined in helper.rs?

**Review by Flouse — CHANGES_REQUESTED** [review]

File: polyjuice-tests/src/ctx.rs (hunk 1 above)
Comment (magicalne) [reply polyjuice-tests/src/ctx.rs]: Fixed!

File: polyjuice-tests/src/ctx.rs [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+            rollup,
+            polyjuice_validator_code_hash,
+            eth_addr_reg_code_hash,
+            secp_data,
+            secp_data_hash,
+        }
+    }
+}
+
+pub fn build_l2_sudt_script(args: [u8; 32]) -> Script {
```
Comment (Flouse) [inline polyjuice-tests/src/ctx.rs:531]: duplicate with helper::build_l2_sudt_script

File: polyjuice-tests/src/ctx.rs [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+    script_args.extend(&ROLLUP_SCRIPT_HASH);
+    script_args.extend(&args[..]);
+    Script::new_builder()
+        .args(Bytes::from(script_args).pack())
+        .code_hash(SUDT_VALIDATOR_SCRIPT_TYPE_HASH.clone().pack())
+        .hash_type(ScriptHashType::Type.into())
+        .build()
+}
+
+pub fn build_eth_l2_script(args: &[u8; 20]) -> Script {
```
Comment (Flouse) [inline polyjuice-tests/src/ctx.rs:542]: duplicate with helper:: build_eth_l2_script

**Review by Flouse — CHANGES_REQUESTED** [review]

**Review by Flouse — APPROVED** [review]

File: devtools/ci/integration-test.sh [hunk 4]
```diff
@@ -6,6 +6,7 @@ PROJECT_ROOT=$(dirname $(dirname $SCRIPT_DIR))
 TESTS_DIR=$PROJECT_ROOT/polyjuice-tests
 DEPS_DIR=$PROJECT_ROOT/integration-test
 GODWOKEN_DIR=$DEPS_DIR/godwoken
+ETHEREUM_TEST_DIR=$DEPS_DIR/test
```
Comment (Flouse) [inline devtools/ci/integration-test.sh:9]:
```suggestion
ETHEREUM_TEST_DIR=$DEPS_DIR/ethereum-tests
```

File: polyjuice-tests/tests/ethereum_test.rs [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+use serde::Deserialize;
+use std::{
+    collections::{BTreeMap, HashMap},
+    convert::TryInto,
+    fs, io,
+    path::{Path, PathBuf},
+    u128,
+};
+
+const TEST_CASE_DIR: &'static str = "../integration-test/test/GeneralStateTests/VMTests/";
```
Comment (Flouse) [inline polyjuice-tests/tests/ethereum_test.rs:13]:
```suggestion
const TEST_CASE_DIR: &'static str = "../integration-test/ethereum-tests/GeneralStateTests/VMTests/";
```

**Review by Flouse — APPROVED** [review]

**Pull request merged**